cmd/client-s3.go

```diff
@@ -989,9 +996,17 @@ func isGoogle(host string) bool {
 }
 
 // Figure out if the URL is of 'virtual host' style.
-// Currently only supported hosts with virtual style
-// are Amazon S3 and Google Cloud Storage.
-func isVirtualHostStyle(host string) bool {
+// Use lookup from config to see if dns/path style look
+// up should be used. If it is set to "auto", use virtual
+// style for supported hosts such as Amazon S3 and Google
+// Cloud Storage. Otherwise, default to path style
+func isVirtualHostStyle(host string, lookup minio.BucketLookupType) bool {
+	if lookup == minio.BucketLookupDNS {
+		return true
+	}
+	if lookup == minio.BucketLookupAuto {
```

Comment:
This doesn't follow isVirtualHostStyle comment, if this is auto we should continue to the next line and decide if this is Amazon or Google etc.. and do not return immediately

Reply:
fixed.